dev_guide/builds.adoc

```diff
@@ -519,9 +531,42 @@ For example, defining a custom HTTP proxy to be used during build:
 ----
 ====
 
+[[source-code]]
+
+== Git Repository Source Options
+
+The source code location is one of the required parameters for the
```

Comment:
today that is true.  but we can remove that restriction.

obviously you're not going to get very far with an s2i-type build that only has a dockerfile as an input source, but i'm willing to let users shoot themselves in the foot with that rather than complicate our logic.  "what did you think was going to happen?"

(and in theory someone could write an s2i builder image that consumes just a dockerfile as the input. not sure what it would do, but technically it's legal)